tests/ui/autoref-autoderef/deref-ambiguity-becomes-nonambiguous.stderr

```diff
@@ -5,7 +5,7 @@ LL |     let var_fn = Value::wrap();
    |         ^^^^^^
 ...
 LL |     let _ = var_fn.clone();
-   |                    ----- type must be known at this point
+   |             ------ type must be known at this point
```

Comment:
all the changes in tests code are pretty good, I'm not sure whether there is some other way to get the self expr's span, so many fn signature need to change for the new extra parameter.

Reply:
c682a70476dc6854e0ad9ab493bb207fcfb5e542 This, pipeline dependent, looks to do the trick